src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticateAndRedirectHandler.cs

```diff
@@ -132,6 +132,10 @@ await AuthenticationHelper.TrySetDigestAuthToken(request, currentCredential, dig
                     (HttpUtilities.IsSupportedSecureScheme(request.RequestUri.Scheme) && HttpUtilities.IsSupportedSecureScheme(location.Scheme));
                 if (!allowed)
                 {
+                    if (NetEventSource.IsEnabled)
+                    {
+                        NetEventSource.Info(this, $"Insecure https to http redirect from {request.RequestUri} to {location} blocked.");
```

Comment:
Unlike the other two, this one is fine, as it's guarded by the NetEventSource.IsEnabled check.  The costs will only be incurred when tracing is enabled.